tests/test-users-api.js

```diff
@@ -0,0 +1,236 @@
+            const result = await pool.query('SELECT public."InsertUser"($1, $2, $3, $4, $5, $6, $7) as new_id', 
+            [birthdate, signupdate, name, cmnd, email, gender, expirydate])
+            var new_user_id = result.rows[0].new_id;
+            console.log("id: " + new_user_id)
+            const response = await axios.get( `${process.env.API_URL}/users/${new_user_id}`);
+            let apiUser = response.data[0];
+            const resultFromData = await pool.query('SELECT * FROM users WHERE id = $1', [new_user_id])
+
+            const dbUser = resultFromData.rows[0];
+            assert.strictEqual(apiUser.id, dbUser.id);
+            assert.strictEqual(apiUser.name, dbUser.name);
+            assert.strictEqual(apiUser.cmnd, dbUser.cmnd);
+            assert.strictEqual(apiUser.gender, dbUser.gender);
+            assert.strictEqual(apiUser.birthdate, dbUser.birthdate.toISOString());
+            assert.strictEqual(apiUser.email, dbUser.email);
+            assert.strictEqual(apiUser.signupdate, dbUser.signupdate.toISOString());
+            assert.strictEqual(apiUser.expirydate, dbUser.expirydate.toISOString());
+            
+            assert.strictEqual(response.status, 200);
+            await pool.query('CALL public."DeleteUser" ($1)', [new_user_id])
+        })
+        it('3. should return status 500 and State will be error when called with incorrect id format', async() => {
+            var new_user_id;
+            const result = await pool.query('SELECT public."InsertUser"($1, $2, $3, $4, $5, $6, $7) as new_id', 
+            [birthdate, signupdate, name, cmnd, email, gender, expirydate])
+            new_user_id = result.rows[0].new_id + '1';
+            console.log("id: " + new_user_id)
+            await axios.get( `${process.env.API_URL}/users/${new_user_id}`).catch(error=>{
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+            await pool.query('CALL public."DeleteUser" ($1)', [result.rows[0].new_id])
+        })
+        it('4. should return status 500 and response will be null when called with incorrect id\'s value', async() => {
+            var new_user_id;
+            const result = await pool.query('SELECT public."InsertUser"($1, $2, $3, $4, $5, $6, $7) as new_id', 
+            [birthdate, signupdate, name, cmnd, email, gender, expirydate])
+            new_user_id = result.rows[0].new_id.substring(0,36) + '1'
+            console.log("id: " + new_user_id)
+            await axios.get( `${process.env.API_URL}/users/${new_user_id}`).catch(error=>{
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
```

Comment:
Description state "response will be null" but the response in code is compared to the not null value.

Reply:
I fixed it

Reply:
Please review the purpose of this test case and how you arrange it. Wrong arrange lead to wrong assert, and you've just changed the description following that wrongness.